Assets/Scripts/DeviceCommunicator.cs

```diff
@@ -0,0 +1,26 @@
+using System.Runtime.InteropServices;
+using System.Text;
+using UnityEngine;
+using Zenject;
+
+public class DeviceCommunicator : MonoBehaviour {
+
+	[Inject]
+	CodeExecutor codeExecutor;
+
+	public void Listen(string protobufInstructionsString)
+	{
+		byte[] protobufInstructions = Encoding.Default.GetBytes (protobufInstructionsString);
+		Code code = Code.Parser.ParseFrom (protobufInstructions);
+
+		codeExecutor.Run (code);
+	}
+
+	[System.Runtime.InteropServices.DllImport("__Internal")]
```

Comment:
I changed to this approach after reading [this article](https://github.com/jiulongw/swift-unity/issues/14) which was published mid-November laster year. It seems to do what we want, but in a much more straight forward way.